session/session.go

```diff
@@ -2049,6 +2048,12 @@ func (s *session) ExecuteStmt(ctx context.Context, stmtNode ast.StmtNode) (sqlex
 		sessionExecuteCompileDurationGeneral.Observe(durCompile.Seconds())
 	}
 	s.currentPlan = stmt.Plan
+	if execStmt, ok := stmtNode.(*ast.ExecuteStmt); ok {
+		if execStmt.Name == "" {
+			// for exec-stmt on bin-protocol, ignore the plan detail in `show process` to gain performance benefits.
+			s.currentPlan = nil
```

Comment:
This original implement(removed by this PR) doesn't set `s.currentPlan`, so setting it to `nil` here can make it compatible with the original.
And I guess why the author of `s.currentPlan` didn't set it on the 1binary-proto exec-stmt1 code path is that he doesn't know there is an extra code path that needs to set it as well, and this is why we want to merge these 2 code paths into 1.

Reply:
`execStmt.Name == ""` means it came from bin-protocol?